api_types.ts

```diff
@@ -22,7 +22,7 @@ export type IsoLanguageCode = string;
 /** ISO (four letter) code of a script. */
 export type IsoScriptCode = string;
 
-/** ISO 8601 `YYYY-MM-DD` date, can be partial (`YYYY-MM-DD` or `YYYY`). */
+/** ISO 8601 `YYYY-MM-DD` date, can be partial (`YYYY-MM-DD` or `YYYY`) or empty. */
```

Comment:
> No rule without an exception: begin and end date of date periods can be null 😁
>
> [...] in some places where this type alias is used it is null (date period) or optional instead of being empty.
>
> The case where it is optional is "first-release-date" of recordings, which is inconsistent with "first-release-date" of release groups where it can only be empty.

Thanks, I've tried to document those issues in [MBS-13519](https://tickets.metabrainz.org/browse/MBS-13519).